fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexer.java

```diff
@@ -823,6 +828,92 @@ public void buildIndex() {
         asyncToSync(buildIndexAsync());
     }
 
+
+    /**
+     * Build an index across multiple transactions in parallel (if needed).
+     * <p>
+     * This method assumes it need to mark the index as readable after building the index. But it will not mark if the
+     * index is built in parallel. It is the caller's responsibility to check all subtasks are completed and then mark
+     * the index as readable.
```

Comment:
This contract feels confusing -- in one circumstance, it fully builds the index and marks it readable.  In another, it dispatches to something that may or may not successfully build the pieces and may or may not eventually mark the index as readable.   I think it would be cleaner to simply have this method compute the set of ranges to be built and leave the determination of what to do with them to the caller.